docs/cluster-management/running-locally.md

```diff
@@ -20,3 +21,108 @@ You will be prompted to enter your Client ID and Client Secret. Afterwards, type
+
+### Custom Docker Images
+
+Since it's powered by Docker, you can determine which images to use for it. SD-in-a-Box (and Screwdriver as a whole) uses the following Docker images:
+
+* [screwdrivercd/screwdriver](https://hub.docker.com/r/screwdrivercd/screwdriver) - API. The main engine of the CI/CD cluster.
+* [screwdrivercd/ui](https://hub.docker.com/r/screwdrivercd/ui) - UI. To pleasantly interact with Screwdriver.
+* [screwdrivercd/store](https://hub.docker.com/r/screwdrivercd/store) - Artifact repository. Responsible for artifacts like build logs & templates
+
+Here is a snippet of the `docker-compose.yml` file
+
+```
+version: '2'
+services:
+  api:
+    image: screwdriver-cd/screwdriver:stable
+    . . .
+  ui:
+    image: screwdriver-cd/ui:stable
+    . . .
+  store:
+    image: screwdriver-cd/store:stable
+    . . .
+```
+
+You can make a local Docker image to use instead of one of these.
+
+We could do a blog post on locally building a docker image that extends one of the Screwdriver ones.
+
+To start up the SD-in-a-Box, execute the following command
+
+```
+$ docker-compose -p screwdriver up
+```
+
+### Local Development Instances
+
+If you plan on making adjustments to a specific Screwdriver component, you can choose to replace a component with your development instance. This will give you a good idea on how it affects the other Screwdriver components before submitting it via Pull Request.
+
+#### General configuration
+
+One important thing to note is that your `docker-compose.yml` will have all the components configured by I.P. address (as opposed to `localhost`). The following features will cease to work if you choose to use `localhost` instead of an I.P. address:
+
+* Builds will not start locally
+
+#### Configuring the UI
+
+You can choose to use a local development instance of the UI.
+
+In development mode, the UI hosts itself on port `4200` and assumes the API is served locally on `8080` by default. You would need to modify the UI's `config/environment.js` file to point to your local Screwdriver cluster, specifcally the API. This can be done by modifying the value right before the `return ENV;` statement.
+
+The following is a snippet that highlights the change you would make in the `config/environment.js`
+
+```
+ . . .
+ ENV.APP.SDAPI_HOSTNAME = 'http://11.22.33.44:8080';  // 8080 is the default. You can also change this
+ return ENV;
+```
+
+The following snippet highlights the `docker-compose.yml` values that need to be modified to use your local UI instance along with the SD-in-a-box cluster.
+
+```yaml
+version: '2'
+services:
+  api:
+    . . .
+    ports:
+      - 8080:80    # UI default port for API is 8080. This can be changed according to the value you set in config/environment.js
+    environment:
+      URI: http://11.22.33.44:8080             # Can't remember. Some way to tell the server what it's origin name is?
```

Comment:
Missed that. It's the value that the Launcher receives so it knows where to update the build progress, logs, and stuff.